### PR TITLE
CI sync test versions

### PR DESCRIPTION
**Pull request by sshine**
This PR further addresses #2.

- CI: Move testing of all exercises to separate script: By separating each step of the CI run script into separate components, we can use those components manually before pushing as part of our workflow.

- `bin/check_exercise_versions`: Add a CI script that checks that `.meta/version` contains a version that is a prefix of `set version ...` in the test file for a given exercise. This ensures that exercise versions are in sync, since we have decided in #10 to keep the canonical version in `.meta/version` and an extra 4th component in the test file.

**TODO:** A CI script that fails if changes are made to test.tcl and 4th component isn't bumped. (I've been working on it, but it's not completely done yet.)

### CODE REVIEW

File: bin/check_exercise_versions [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        cd ..
+        continue
+    fi
+
+    META_VERSION=$(cat .meta/version)
+    TEST_VERSION=$(echo 'source test.tcl; if {[info exists version]} {puts $version}' | tclsh)
+
+    # Make sure each exercise has a 'set version ...' statement.
+
+    if [ -z $TEST_VERSION ]; then
```
Comment (glennj) [inline bin/check_exercise_versions]: Always quote variables, especially inside brackets: if the TEST_VERSION variable is empty, you get `if [ -z  ]` and the `[` command with only a single argument returns "true" if the argument is not empty: "-z" is not empty
```sh
    if [ -z "$TEXT_VERSION" ]; then
```

File: bin/check_exercise_versions [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            cd ..
+            continue
+            ;;
+    esac
+
+    echo " - $exercise has version $TEST_VERSION"
+    cd ..
+done
+
+if [ $FAILED ]; then
```
Comment (glennj) [inline bin/check_exercise_versions]: `if [ "$FAILED" -eq 1 ]; then`

File: bin/check_exercise_versions [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        FAILED=1
+        err " - $exercise has no 'set version ...' statement!"
+        cd ..
+        continue
+    fi
+
+    # Make sure .meta/version is a prefix of 'set version ...'
+
+    case $TEST_VERSION in
+        $META_VERSION.*)
```
Comment (glennj) [inline bin/check_exercise_versions]: You want to switch these up: META_VERSION is the more specific:
```sh
case "$META_VERSION" in
    "$TEST_VERSION".*) : ;;  # should have at least one command
    *) FAILED=1; ... ;;
esac
```

Comment (glennj) [reply bin/check_exercise_versions]: Or perhaps:
```sh
if [ "$TEST_VERSION" = "${META_VERSION%.*}" ]; then    # strip off the *last* dot and following chars
    : OK
else
    FAILED=1
    ...
fi
```

Comment (sshine) [reply bin/check_exercise_versions]: No, `$TEST_VERSION` is the more specific holding a 4th component with `$META_VERSION` holding just the canonical version with three components. At least, it is in the two exercises I've committed so far. Do you think it is too confusing to keep two different version numbers? We could just simplify so they're both the canonical version.

**Comment by sshine** [comment]
I have reverted version numbers so that both have 3 components for now.

Let's have a further discussion in #10 if necessary.

**Pull request merged**